src/main/java/ch/njol/skript/expressions/ExprReversedList.java

```diff
@@ -46,22 +48,23 @@
 		Skript.registerExpression(ExprReversedList.class, Object.class, ExpressionType.COMBINED, "reversed %objects%");
 	}
 
-	@SuppressWarnings("null")
+	@SuppressWarnings("NotNullFieldNotInitialized")
 	private Expression<?> list;
 
 	@Override
-	@SuppressWarnings({"null", "unchecked"})
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, SkriptParser.ParseResult parseResult) {
-		list = exprs[0].getConvertedExpression(Object.class);
-		return list != null;
+		list = LiteralUtils.defendExpression(exprs[0]);
+		return LiteralUtils.canInitSafely(list);
 	}
 
 	@Override
 	@Nullable
 	protected Object[] get(Event e) {
-		List<Object> reversed = Arrays.asList(list.getAll(e).clone());
+		Object[] inputArray = list.getArray(e).clone();
+		List<?> reversed = Arrays.asList(inputArray);
```

Comment:
I would rather this to be done manually rather than creating a list for it, like this:
```java
Object[] input = list.getArray(e).clone();
return reversed(input);

/* This is in a better code base would've had methods dealing with primitives as well 
  to avoid boxing/unboxing but since Skript erases type by using generics, 
  it *shouldn't* be a concern for us
*/
private <T> T reversed(T[] array) {
    for (int i = 0; i < array.length / 2; i++) {
        int curr = array[i];
        int reverse = array.length - i - 1;
        array[i] = array[reverse];
        array[reverse] = temp;
    }
    return array;
}
```
While this expression wouldn't be able to generate a huge bottleneck by itself, cloning an array and then making a List of the same size is quite not efficient (even for a shallow clone) considering the impl that reverses the array to be quite simple.

I know this was not the point of the PR but it'd be good to revisit it while we're modifying these classes.